File: 2023/py_solutions/easy/sol_242.py
# 242. Valid Anagram
# https://leetcode.com/problems/valid-anagram/description/

class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        sorted_s=sorted([ord(n) for n in list(s)])
        sorted_t=sorted([ord(n) for n in list(t)])
        return sorted_s == sorted_t

# hashmap solution
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        count_s, count_t = {}, {}
        for i in range(len(s)):
            count_s[s[i]] = 1 + count_s.get(s[i], 0)
            count_t[t[i]] = 1 + count_t.get(t[i], 0)
        for char in count_s:
            print(char)
            if count_s[char] != count_t.get(char):
                return False
        return True

File: 2023/js_solutions/easy/sol242.js
// 242. Valid Anagram
// https://leetcode.com/problems/valid-anagram/description/

var isAnagram = function(s, t) {
    let sOrd = []
    let tOrd = []
    if (s.length !== t.length){
        return false
    }
    for (let i=0; i < s.length; i++){
        sOrd.push(s.charCodeAt(i))
        tOrd.push(t.charCodeAt(i))
    }
    sOrd.sort((a,b)=>a-b)
    tOrd.sort((a,b)=>a-b)
    console.log("s", sOrd, "t", tOrd)
    for (let i=0; i < s.length; i++){
        if (sOrd[i] != tOrd[i]) {
            console.log("i", i)
            return false
        }
    }
    return true
};
